### PR TITLE
fix(DnD): [FOR-431] remove clean residue (obsolete) on DnD question in edit-form

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
An old problem on form edition was the apparition of a "phantom question", a kind of question duplicated but not really existing (disapear on refresh). It was happenning when drag and drop a question from a section outside of it.
A fix was made to delete this unwanted effect.

But since a fix was made on the drag and drop this effect doesn't happen anymore and the fix is now obsolete and created another problem.

So here, we removed that obsolete fix.

## Issue ticket number and link
FOR-431 : https://jira.support-ent.fr/browse/FOR-431

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: common/src/main/resources/ts/utils/formElement.ts [hunk 1]
```diff
@@ -104,14 +104,14 @@ export class FormElementUtils {
 
     // Drag and drop
 
-    static onEndDragAndDrop = async (evt: any, formElements: FormElements) : Promise<boolean> => {
-        let scopElem = angular.element(evt.item.firstElementChild.firstElementChild).scope().vm;
-        let itemId = scopElem.question ? scopElem.question.id : scopElem.section.id;
+    static onEndDragAndDrop = async (evt: any, formElements: FormElements) : Promise<void> => {
+        let scopeElem = angular.element(evt.item.firstElementChild.firstElementChild).scope().vm;
+        let itemId = scopeElem.question ? scopeElem.question.id : scopeElem.section.id;
```
Comment (BastienBenard) [inline common/src/main/resources/ts/utils/formElement.ts:109]: typage

File: formulaire/src/main/resources/public/ts/controllers/form-editor.ts [hunk 2]
```diff
@@ -638,15 +638,11 @@ export const formEditorController = ng.controller('FormEditorController', ['$sco
                         document.querySelector('header').style.pointerEvents = 'none';
                     },
                     onEnd: async function (evt) {
-                        let cleanResidue = await FormElementUtils.onEndDragAndDrop(evt, vm.formElements);
+                        await FormElementUtils.onEndDragAndDrop(evt, vm.formElements);
                         document.querySelector('header').style.removeProperty('pointer-events');
                         $scope.safeApply();
 
                         await vm.$onInit();
```
Comment (BastienBenard) [inline formulaire/src/main/resources/public/ts/controllers/form-editor.ts:645]: Il ne faudrait pas un try catch pour les 2 await ?

**Review by BastienBenard — APPROVED** [review]

File: formulaire/src/main/resources/public/ts/controllers/form-editor.ts (hunk 2 above)
Comment (qperie) [reply formulaire/src/main/resources/public/ts/controllers/form-editor.ts:645]: Ca appelle des service dans lesquels y'a déjà les try-catch

File: common/src/main/resources/ts/utils/formElement.ts [hunk 3]
```diff
@@ -104,27 +104,27 @@ export class FormElementUtils {
...
-        let oldSection = oldNestedContainerId ? (formElements.all.filter(e => e instanceof Section && e.id === oldNestedContainerId)[0]) as Section: null;
-        let item = null;
-        if (scopElem.section) {
-            item = formElements.all.filter(q => q.id === itemId)[0] as Section;
+    static onEndDragAndDrop = async (evt: any, formElements: FormElements) : Promise<void> => {
+        let scopeElem: any = angular.element(evt.item.firstElementChild.firstElementChild).scope().vm;
+        let itemId: number = scopeElem.question ? scopeElem.question.id : scopeElem.section.id;
+        let newNestedSectionId: number = evt.to.id.split("-")[1] != "0" ? parseInt(evt.to.id.split("-")[1]) : null;
+        let oldNestedContainerId: number = evt.from.id.split("-")[1] != "0" ? parseInt(evt.from.id.split("-")[1]) : null;
+        let oldSection: Section = oldNestedContainerId ? (formElements.all.filter(e => e instanceof Section && e.id === oldNestedContainerId)[0]) as Section : null;
```
Comment (BastienBenard) [inline common/src/main/resources/ts/utils/formElement.ts:112]: est ce que e ne peux pas etre typé en FormElement ?

File: common/src/main/resources/ts/utils/formElement.ts [hunk 4]
```diff
@@ -104,27 +104,27 @@ export class FormElementUtils {
...
+        let item: any = null;
+        if (scopeElem.section) {
+            item = formElements.all.filter(e => e.id === itemId)[0] as Section;
```
Comment (BastienBenard) [inline common/src/main/resources/ts/utils/formElement.ts:115]: est ce que e ne peux pas etre typé en FormElement ?

File: common/src/main/resources/ts/utils/formElement.ts [hunk 5]
```diff
@@ -104,27 +104,27 @@ export class FormElementUtils {
...
-        let item = null;
-        if (scopElem.section) {
-            item = formElements.all.filter(q => q.id === itemId)[0] as Section;
+    static onEndDragAndDrop = async (evt: any, formElements: FormElements) : Promise<void> => {
+        let scopeElem: any = angular.element(evt.item.firstElementChild.firstElementChild).scope().vm;
+        let itemId: number = scopeElem.question ? scopeElem.question.id : scopeElem.section.id;
+        let newNestedSectionId: number = evt.to.id.split("-")[1] != "0" ? parseInt(evt.to.id.split("-")[1]) : null;
+        let oldNestedContainerId: number = evt.from.id.split("-")[1] != "0" ? parseInt(evt.from.id.split("-")[1]) : null;
+        let oldSection: Section = oldNestedContainerId ? (formElements.all.filter(e => e instanceof Section && e.id === oldNestedContainerId)[0]) as Section : null;
+        let item: any = null;
```
Comment (BastienBenard) [inline common/src/main/resources/ts/utils/formElement.ts:113]: item peut etre type en Section | Question
Mais si possible ce serait mieux de faire 2 variables

File: common/src/main/resources/ts/utils/formElement.ts (hunk 3 above)
Comment (qperie) [reply common/src/main/resources/ts/utils/formElement.ts:112]: C'est un type Section, aucune raison de le typer comme FormElement

**Pull request merged**